experimental/dds/tree2/src/feature-libraries/chunked-forest/codec/schemaBasedEncoding.ts

```diff
@@ -130,3 +132,33 @@ function valueShapeFromSchema(schema: ValueSchema | undefined): undefined | Enco
 			unreachableCase(schema);
 	}
 }
+
+export function makeSchemaCompressedCodec(
+	{ jsonValidator: validator }: ICodecOptions,
+	schema: SchemaData,
+	policy: FullSchemaPolicy,
+): IJsonCodec<ITreeCursorSynchronous, string> {
+	const versionedValidator = validator.compile(Versioned);
+	const formatValidator = validator.compile(EncodedChunk);
+	return {
+		encode: (data: ITreeCursorSynchronous) => {
+			const encoded = schemaCompressedEncode(schema, policy, data);
+			assert(versionedValidator.check(encoded), "Encoded schema should be versioned");
+			assert(formatValidator.check(encoded), "Encoded schema should validate");
+			return JSON.stringify(encoded);
```

Comment:
I'd avoid stringifying here and just make the return type `IJsonCodec<ITreeCursorSynchronous>`. The current implementation won't support IFluidHandles, but if you avoid using JSON.stringify it likely will once integrated into ForestSummarizer (that uses an IFluidSerializer to serialize the result of `jsonableTreeFromCursor`, and the same approach will work on your encoded data)

Reply:
similar case for all the other codecs. in general avoid JSON.stringify in encoding logic, it usually causes issues with handles.

Reply:
Removed JSON.stringify and updated codec return type.